backend/de.metas.util/src/main/java/de/metas/util/time/DurationUtils.java

```diff
@@ -35,10 +38,32 @@
 @UtilityClass
 public class DurationUtils
 {
-	public static Duration toDuration(@NonNull final BigDecimal durationBD, @NonNull final TemporalUnit unit)
+	private static final List<TemporalUnit> supportedTemporalUnits = ImmutableList.of(ChronoUnit.YEARS, ChronoUnit.MONTHS, ChronoUnit.DAYS, ChronoUnit.HOURS, ChronoUnit.MINUTES, ChronoUnit.SECONDS, ChronoUnit.NANOS);
+
+	public static Duration toWorkDuration(final @NonNull BigDecimal durationBD, final @NonNull TemporalUnit unit)
+	{
+		BigDecimal duration = durationBD;
+		TemporalUnit currentUnit = unit;
+		long durationLong = -1;
+		while (durationLong < 0)
```

Comment:
i'm a bit nervous about this while loop. what we we end up with durationLong < 0 and it never changes anymore?
why not instead have a for-loop 
or a counter? we can anyways reasonably just iterate as often as we have supportedTemporalUnits, right?

Reply:
case in point: 
![image](https://user-images.githubusercontent.com/1264789/124919356-6f77c080-dff6-11eb-9045-1d3b9ad11b74.png)

Reply:
Changed to prevent negative duration from being passed in, and also added a second hard stop condition regarding the ChronoUnit.